lib/class/catalog.class.php

```diff
@@ -1460,14 +1460,22 @@ public function gather_art($songs = null, $videos = null)
         // Prevent the script from timing out
         set_time_limit(0);
 
+        $gather_song_art = AmpConfig::get('gather_song_art') ?: false;
+
         $search_count = 0;
         $searches     = array();
         if ($songs == null) {
             $searches['album']  = $this->get_album_ids('art');
             $searches['artist'] = $this->get_artist_ids('art');
+            if ($gather_song_art) {
+                $searches['song'] = $this->get_songs();
+            }
         } else {
             $searches['album']  = array();
             $searches['artist'] = array();
+            if ($gather_song_art) {
+                $searches['song'] = array();
+            }
             foreach ($songs as $song_id) {
                 $song = new Song($song_id);
                 if ($song->id) {
```

Comment:
The objects were already created/returned on line 1471. To be consistent, I suggest that song IDs be returned and objects created here.